be/src/vec/aggregate_functions/aggregate_function_regr_union.h

```diff
@@ -45,12 +45,14 @@ struct AggregateFunctionRegrData {
     Float64 sum_y {};
     Float64 sum_of_x_mul_y {};
     Float64 sum_of_x_squared {};
+    Float64 sum_of_y_squared {};
```

Comment:
`sum_of_y_squared` is only needed for `regr_syy`, in other situations, this field is not necessary.
so we should use template to make our implementation more efficient.

Reply:
done